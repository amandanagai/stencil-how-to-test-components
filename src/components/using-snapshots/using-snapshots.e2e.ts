import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('using-snapshots e2e', () => {
  describe('rendering', () => {
    let page: E2EPage;
    let element: E2EElement;

    beforeEach(async () => {
      page = await newE2EPage({
        html: `
        <using-snapshots></using-snapshots>
      `
      });
      element = await page.find('using-snapshots');
    });

    it('should work with both the first and the last name', async () => {
      await element.setProperty('first', 'Peter');
      await element.setProperty('last', 'Parker');
      await page.waitForChanges();

      expect(element.innerHTML).toMatchSnapshot();
    });
  });
});
