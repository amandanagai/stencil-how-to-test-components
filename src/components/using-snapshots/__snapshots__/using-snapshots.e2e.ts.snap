// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using-snapshots e2e rendering should work with both the first and the last name 1`] = `"<div>Hello, my name is Peter Parker</div>"`;
